core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -192,8 +212,11 @@ public SnapshotInfo snapshot(SnapshotId snapshotId) {
      * @param listener snapshot creation listener
      */
     public void createSnapshot(final SnapshotRequest request, final CreateSnapshotListener listener) {
-        final SnapshotId snapshotId = new SnapshotId(request.repository(), request.name());
-        validate(snapshotId);
+        final String repositoryName = request.repositoryName;
+        final String snapshotName = request.snapshotName;
+        validate(repositoryName, snapshotName);
+        final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID()); // new UUID for the snapshot
```

Comment:
It's used 1 time in production code and 16 times in the tests... For most of these tests, we also chose a random snapshot name and when chose a specific snapshot name we can as well just use a fixed uuid. I'm not sure that warrants adding a constructor...